src/Components/Blazor/Server/src/MonoDebugProxy/BlazorMonoDebugProxyAppBuilderExtensions.cs

```diff
@@ -52,6 +54,71 @@ public static void UseBlazorDebugging(this IApplicationBuilder app)
             });
         }
 
+        private static void UseVisualStudioDebuggerConnectionRequestHandlers(this IApplicationBuilder app)
+        {
+            // Unfortunately VS doesn't send any deliberately distinguishing information so we know it's
+            // not a regular browser or API client. The closest we can do is look for the *absence* of a
+            // User-Agent header. In the future, we should try to get VS to send a special header to indicate
+            // this is a debugger metadata request.
+            app.Use(async (context, next) =>
+            {
+                var request = context.Request;
+                var requestPath = request.Path;
+                if (requestPath.StartsWithSegments("/json")
+                    && !request.Headers.ContainsKey("User-Agent"))
+                {
+                    if (requestPath.Equals("/json", StringComparison.OrdinalIgnoreCase) || requestPath.Equals("/json/list", StringComparison.OrdinalIgnoreCase))
+                    {
+                        var debuggerHost = "http://localhost:9222";
```

Comment:
Should this be an option?

Reply:
Hmm, hypothetically yes, but it might be slightly more complicated than that. The only config type I can find (`BlazorConfig`) for Blazor is not injected VIA DI in the normal way, but created where needed via `BlazorConfig.Read(assemblyPath)`, meaning we have to either start passing the assemblyPath to `UseBlazorDebugging` (a public API, so a breaking change) or re-architect BlazorConfig to work in a more standardized way (also probably a breaking change since we'd remove the `assemblyPath` property from anywhere using it for BlazorConfig).

I can do that work, but frankly after the needed API review and solicitation of opinions it seems like more than we should bite off if we're going to get this done in the timeframe @danroth27 wants. I suggest filing an issue to address BlazorConfig as a whole (currently that file is a non-standard format that we're dealing with ad-hoc, and there are comments all around it calling for that to change), including an option for DebugPort.

CC @SteveSandersonMS for thoughts and opinions here.

Reply:
Any config option for Blazor WebAssembly that supports running in the dev server cannot use similar patterns to server-side ASP.NET Core, because when you're running in the dev server, your only .NET code is your wasm code. You have no .NET running on the server that you (the application developer) wrote directly.

So if this is to be configurable, it would need to be via an MSBuild property, and we would need to update the build targets to write out that config setting into `BlazorConfig`, and then the correct way to access it at runtime is indeed to use `BlazorConfig.Read(assemblyPath)`. It doesn't seem like there really is any alternative to "re-architect BlazorConfig to work in a more standardized way", but please let me know if you think I'm missing understanding of how it could be made more standardized.

At this point it would be worth considering adding an extra overload of `UseBlazorDebugging` that takes some parameter we can use to identify the client-side app. For example, it could be `UseBlazorDebugging<TClientApp>` for consistency with `app.UseClientSideBlazorFiles<TClientApp>`. Then we're safe adding whatever debug config options we want in the future.

Anyone who uses the older `UseClientSideBlazorFiles()` would have to get default behaviors for these configurable features.

@rynowak Any other ideas or preferences?

Reply:
We could accept an options type (accepted via the ctor) that allows configuring the middleware? That's usually how you configure middlewares. Users could configure it by invoking `services.Configure<BlazorDebuggingOptions>(options => ...)` in their `ConfigureServices`.

That should work, no?

Reply:
Not for Blazor standalone. There's nowhere for that code to go.

Reply:
Ahh yes, you're right. We should just park this for now. There's also this other issue - https://github.com/aspnet/AspNetCore/issues/13247 - which involves figuring out how to configure the dev-server defaults, perhaps we could solve these two together.